src/stirling/source_connectors/socket_tracer/protocols/common/event_parser.h

```diff
@@ -129,6 +129,7 @@ ParseResult ParseFrames(message_type_t type, DataStreamBuffer* data_stream_buffe
   VLOG(1) << absl::Substitute("Parsed $0 new frames", frames->size() - prev_size);
 
   // Match timestamps with the parsed frames.
+  uint64_t prev_timestamp_ns = 0;
   for (size_t i = 0; i < result.frame_positions.size(); ++i) {
```

Comment:
I've added a field to the datastream buffer `StatusOr<uint64_t> prev_timestamp_ns_;`.  Whenever a non-monotonic timestamp is encountered, we log a warning, and return the previous timestamp.

It may be cleaner to move the error handling in `event_parser.h` to `data_stream_buffer.h`.
```cpp
LOG_IF(ERROR, !timestamp_ns_status.ok()) << timestamp_ns_status.ToString();
```
Currently `GetTimestamp` simply returns a non-OK statusOr to be handled by the caller.
```cpp
if (!timestamp_ns_status.ok()) {
  return timestamp_ns_status;
}
```
Curious to hear your thoughts on this.

Reply:
Another question is whether we want to `ValueOr(0)` here, as we do in `event_parser`
```cpp
uint64_t current_timestamp_ns = timestamp_ns_status.ValueOr(0);
uint64_t prev_timestamp_ns = GetPrevTimestamp().ValueOr(0);
```

Reply:
I'm fine with the error handling in the current code, however, I could see a case for moving it from `event_parser.h` to `data_stream_buffer.h`. As far as I can tell, the only caller of `GetTimestamp` (outside of tests) is the event parser. So we could keep the existing `ValueOr(0)` semantic entirely within the buffer. No strong opinion one way or the other 

Regarding the `ValueOr(0)`, see my comment about possibly using `ConsumeValueOrDie()`.